### DISCUSSION TITLE
Authentication in Strapi using next-auth library

### DISCUSSION THREAD

**Discussion by alfieqashwa** [strapi/strapi#7477] [Legacy Topics]
How to create authentication using [next-auth](https://www.npmjs.com/package/next-auth) library?

I've been asking to the creator of next-auth Mr. @iaincollins and he was referencing me there some people were asking the similar questions. [#574](https://github.com/nextauthjs/next-auth/issues/574#issuecomment-673404573) 

I think that would be good idea if you guys collaborate with. I love Strapi. I watched [Alex the Entreprenerd](https://www.youtube.com/watch?v=vGtVSwpOlsM) and assuming that the authentication of Strapi as a framework is still hard-coded, also when reading the [Strapi Official Blog](https://strapi.io/blog/nextjs-react-hooks-strapi-auth-4).

CMIIW, Thank you !

**Comment by MattieBelt** [comment]
Hey @alfieqashwa, FYI:

Yesterday @ghoshnirmalya offered to provide the [strapi/community-content](https://github.com/strapi/community-content) with an article/tutorial about autehntication with Next.js [#123](https://github.com/strapi/community-content/issues/123).

**Comment by derrickmehaffy** [comment]
As I'm not largely familiar with the next-auth lib, was there something specific that is needed from us @alfieqashwa ?

(Thanks @MattieBelt I was gonna link that also and you beat me to it :laughing: )

**Comment by iaincollins** [comment]
@alfieqashwa Thanks for raising this!

@MattieBelt Oh that's great news! For context we have a bunch of folk ask about how to integrate NextAuth.js with Strapi which I think is entirely possible, but I'm not familiar with Strapi so haven't been able to help.

I really appreciate the input and issues @ghoshnirmalya has helped out with on NextAuth.js.

**Comment by alfieqashwa** [comment]
Hey guys. I'm so happy and can't wait for the [tutorial](https://github.com/strapi/community-content/issues/123#issuecomment-673627571). 

Thank you @MattieBelt @derrickmehaffy @iaincollins @ghoshnirmalya for this good news.

**Comment by lauriejim** [comment]
I ping @Mcastres about that thread.

**Comment by oskar-gmerek** [comment]
This is missing here. Good implementation of strapi auth used by next-auth
